Neo-4582: Post API Python Examples on Github

bensont opened the pull request:
- Added the Python example code to the repository
- Updated README.md to explain how to get the examples running

Commit:
Added api client python examples and updated README.md

LogicalChaos commented on README.md:
```diff
@@ -1,2 +1,34 @@
 # storcycle_api
-StorCycle API Examples
+###StorCycle API Examples
```
Need a space between your # and the title, same as below.  And you should use a progression of #, so this one would have 2, not 3, and the python one below should have 3, not 4.

michaelharperSpectra commented on api_examples_final.py:
```diff
@@ -0,0 +1,538 @@
+sleep_time_seconds = 10
+
+# Note: If the example directory tree is not created, these example variables will need to be changed.
+example_directory_root = "C:\\ApiPythonExampleData"
```
might want to make this something like os.path.join(os.path.abspath(os.sep), "ApiPythonExampleData") to make this platform independent

os.path.abspath(os.sep) should return C:\\ on windows and / on linux

michaelharperSpectra commented on api_examples_final.py:
```diff
@@ -0,0 +1,538 @@
+        return ApiProjectScan(
+            description='A description of the scan project',
+            share='Example-Source-Optional-Fields',
+            working_directory='\\WorkingDirectory',
```
similarly hear you may want to use os.path.sep instead of hard coding \\

LogicalChaos commented on README.md:
```diff
@@ -1,2 +1,34 @@
 # storcycle_api
```
Probably want to put in the link to the OpenAPI here with words.

bensont replied:
@LogicalChaos, what do you mean by "the OpenAPI"? A link to the Python examples file or to the swagger documentation?

LogicalChaos commented:
Swagger documentation

> On Jan 5, 2021, at 12:01 PM, Tyler Benson <notifications@github.com> wrote:
> 
> 
> @bensont commented on this pull request.
> 
> In README.md <https://github.com/SpectraLogic/storcycle_api/pull/1#discussion_r552131808>:
> 
> > @@ -1,2 +1,34 @@
>  # storcycle_api
> @LogicalChaos <https://github.com/LogicalChaos>, what do you mean by "the OpenAPI"? A link to the Python examples file or to the swagger documentation?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/SpectraLogic/storcycle_api/pull/1#discussion_r552131808>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ABSQI4RGHYLAIJ3M5SS6T43SYNOZJANCNFSM4VVTC3SQ>.
>

Commit:
Changes requested from PR reviews. Fixed some markdown formatting. Fixed some paths in the examples to not be platform specific. Added a link to the swagger documentation in README.md. Changed the example file name.

LogicalChaos commented on README.md:
```diff
@@ -1,2 +1,37 @@
 # storcycle_api
-StorCycle API Examples
+
+[Swagger Documentation](https://localhost/apidocs/)
```
I'd point them to https://localhost/openapi.yaml, as the link you specify, by default, shows the internal API swagger doc.  I'm not sure how to fix that.

michaelharperSpectra replied:
on every build i've tested https://localhost/apidocs/ points to https://localhost/openapi.yaml

michaelharperSpectra replied:
ex: https://10.2.2.146/apidocs/

LogicalChaos replied:
Good, must be unique to a dev build that it's not quite correct.

RachelTucker commented on README.md:
```diff
@@ -1,2 +1,37 @@
+
+### Python:
+
+Download a client generator plugin for your local IDE and install. Note that while the OpenAPI Generator plugin by Jim Schubert was used to generate the example Python client, any client generator should be adequate.
```
These instructions appear to be hard-coded to assuming an intellij IDE. Since this is for python, did you check that pycharm has the OpenAPI generator plugin?

We should probably include command line instructions as well and a link to the open api website for downloading their library directly.

bensont replied:
@RachelTucker, I have checked that PyCharm has the OpenAPI generator plugin.

I can add command line instructions. To what are you referring with downloading the open api website library directly? I'm not seeing anything on [https://www.openapis.org/](https://www.openapis.org/) for libraries.

LogicalChaos replied:
https://openapi-generator.tech/docs/installation/

bensont replied:
Awesome, I've already included that link, I just misunderstood this context. Thanks!

LogicalChaos commented on README.md:
```diff
@@ -1,2 +1,37 @@
+
+To use the examples, client code must first be generated in the desired language.
+
+### Python:
```
I'd put in a command line example for doing the generation also.  `java -jar openapi/codegen/1283369100_openapi-generator-cli-4.3.1.jar generate -g python -i openapi.yaml -o /tmp/`

bensont replied:
@LogicalChaos, I assume I should also provide the link to [https://openapi-generator.tech/docs/installation/](https://openapi-generator.tech/docs/installation/) to get the client generator?

LogicalChaos replied:
Sounds good

LogicalChaos commented on README.md:
```diff
@@ -1,2 +1,37 @@
+
+### Python:
+
+Download a client generator plugin for your local IDE and install. Note that while the OpenAPI Generator plugin by Jim Schubert was used to generate the example Python client, any client generator should be adequate.
```
```suggestion
Download a client generator plugin for your local IDE and install. Note that while the OpenAPI Generator plugin by Jim Schubert was used within Jet Brains IntelliJ to generate the example Python client, any client generator should be adequate.
```

RachelTucker commented on api_examples.py:
```diff
@@ -0,0 +1,538 @@
+# coding: utf-8
```
```suggestion
#   Copyright 2021 Spectra Logic Corporation. All Rights Reserved.
#   Licensed under the Apache License, Version 2.0 (the "License"). You may not use
#   this file except in compliance with the License. A copy of the License is located at
#
#   http://www.apache.org/licenses/LICENSE-2.0
#
#   or in the "license" file accompanying this file.
#   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
#   CONDITIONS OF ANY KIND, either express or implied. See the License for the
#   specific language governing permissions and limitations under the License.

# coding: utf-8
```
We need to add a copyright

RachelTucker replied:
nm, I need to read further

RachelTucker commented on api_examples.py:
```diff
@@ -0,0 +1,538 @@
+    StorCycle® REST API Copyright © 2020 Spectra Logic Corporation  # noqa: E501
+
+    The version of the OpenAPI document: 0.1
+    Contact: developer@spectralogic.com
```
this is not a real email

LogicalChaos commented:
Yes that’s a real email.  I’m one of the recipients of said email.  Jeff B, maybe Steven E. also.

> On Jan 5, 2021, at 12:36 PM, RachelTucker <notifications@github.com> wrote:
> 
> 
> @RachelTucker commented on this pull request.
> 
> In api_examples.py <https://github.com/SpectraLogic/storcycle_api/pull/1#discussion_r552150339>:
> 
> > @@ -0,0 +1,538 @@
> +# coding: utf-8
> +
> +"""
> +    Spectra Logic Corporation Software
> +
> +    StorCycle® REST API Copyright © 2020 Spectra Logic Corporation  # noqa: E501
> +
> +    The version of the OpenAPI document: 0.1
> +    Contact: developer@spectralogic.com
> this is not a real email
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/SpectraLogic/storcycle_api/pull/1#pullrequestreview-562075238>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ABSQI4QLNBISIQV3CWEMHU3SYNS5NANCNFSM4VVTC3SQ>.
>

RachelTucker commented:
learn something new every day

LogicalChaos commented:
I’m referring to downloading the yaml file directly, rather than just viewing it online.  http://localhost:8080/apidocs/openapi.yaml <http://localhost:8080/apidocs/openapi.yaml> allows you to do that.

> On Jan 5, 2021, at 3:06 PM, Tyler Benson <notifications@github.com> wrote:
> 
> 
> @bensont commented on this pull request.
> 
> In README.md <https://github.com/SpectraLogic/storcycle_api/pull/1#discussion_r552227091>:
> 
> > @@ -1,2 +1,37 @@
>  # storcycle_api
> -StorCycle API Examples
> +
> +[Swagger Documentation](https://localhost/apidocs/)
> +
> +## StorCycle API Examples
> +
> +This repository contains examples for interacting with the StorCycle API. These general examples require either creation of the example directory tree, or adjusting variables to suit the local system.
> +
> +To use the examples, client code must first be generated in the desired language.
> +
> +### Python:
> +
> +Download a client generator plugin for your local IDE and install. Note that while the OpenAPI Generator plugin by Jim Schubert was used to generate the example Python client, any client generator should be adequate.
> @RachelTucker <https://github.com/RachelTucker>, I have checked that PyCharm has the OpenAPI generator plugin.
> 
> I can add command line instructions. To what are you referring with downloading the open api website library directly? I'm not seeing anything on https://www.openapis.org/ <https://www.openapis.org/> for libraries.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/SpectraLogic/storcycle_api/pull/1#discussion_r552227091>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ABSQI4SGHWCX7WFV5XC7IZTSYOENBANCNFSM4VVTC3SQ>.
>

Commit:
Neo-4582: Post API Python Examples on Github:
Added link to view the openapi.yaml file. Added command line generation example and links for getting the open api client generator. Example for running via command line or IDE.
